waffle_utils/video/tools.py

```diff
@@ -0,0 +1,118 @@
+    save_image,
+)
+from waffle_utils.video.defaults import DEFAULT_FRAME_RATE
+
+
+def extract_frames(
+    input_path: Union[str, Path],
+    output_dir: Union[str, Path],
+    frame_rate: int = DEFAULT_FRAME_RATE,
+    verbose: bool = False,
+) -> None:
+    """Extract Frames as Individual Images from a Video File
+
+    Args:
+        input_path (Union[str, Path]): Path to the input video file.
+        output_dir (Union[str, Path]): Path to the output directory where the
+            frame images will be saved.
+        # TODO: Check if it prints the value of `{DEFAULT_FRAME_RATE}` in
+        documentation
+        frame_rate (int, optional): Frame rate of the output images. Defaults
+            to {DEFAULT_FRAME_RATE}.
+        verbose (bool, optional): Whether to print verbose output. Defaults to
+            False.
+    """
+
+    input_path = Path(input_path)
+    output_dir = Path(output_dir)
+
+    # Create output directory if it doesn't exist
+    if not output_dir.exists():
+        output_dir.mkdir(parents=True, exist_ok=True)
+
+    # Extract frames from the video file
+    video_capture = create_video_capture(input_path)
+    fps = video_capture.get(cv2.CAP_PROP_FPS)
+    success, image = video_capture.read()
+    count = 0
+    frame_interval = int(round(fps / frame_rate))
+    while success:
+        count += 1
+        if count % frame_interval == 0:
+            output_path = output_dir / f"{count}.jpg"
+            save_image(output_path, image)
+
+            if verbose:
+                print(f"Extracted frame {count} to {output_path}.")
+
+        success, image = video_capture.read()
+
+    video_capture.release()
+
+
+def create_video(
+    input_dir: Union[str, Path],
+    output_path: Union[str, Path],
+    frame_rate: int = DEFAULT_FRAME_RATE,
+    verbose: bool = False,
+) -> None:
+    """Create a Video File from a Directory of Frame Images.
+
+    Args:
+        input_dir (Union[str, Path]): Path to the input directory containing
+            the frame images.
+        output_path (Union[str, Path]): Path to the output video file.
+        # TODO: Check if it prints the value of `{DEFAULT_FRAME_RATE}` in
+        documentation
+        frame_rate (int, optional): Frame rate of the output video. Defaults
+            to {DEFAULT_FRAME_RATE}.
+        verbose (bool, optional): Whether to print verbose output. Defaults to
+            False.
+    """
+
+    input_dir = Path(input_dir)
+    output_path = Path(output_path)
+
+    # Create output directory if it doesn't exist
+    if not output_path.parent.exists():
+        make_directory(input_dir)
+
+    # Get a sorted list of frame image files
+    frames = sorted(input_dir.glob("*.jpg"))
+
+    # Load the first frame to get dimensions
+    first_frame = load_image(frames[0])
+    height, width, _ = first_frame.shape
+
+    # Initialize video writer with the desired codec, frame rate, and frame
+    # size
+    fourcc = cv2.VideoWriter_fourcc(*"mp4v")
```

Comment:
Is the output video format fixed to MP4? If you add or change later, you will need to modify the code?

Reply:
I added the support video format
- `.mp4`
- `.avi`
- `.wmv`
- `.mov`
- `.flv`
- `.mkv`
- `.mpeg`

Reply:
Done